x/evm/keeper/abci.go

```diff
@@ -44,5 +52,33 @@ func (k *Keeper) EndBlock(ctx context.Context) error {
 	bloom := ethtypes.BytesToBloom(k.GetBlockBloomTransient(infCtx).Bytes())
 	k.EmitBlockBloomEvent(infCtx, bloom)
 
+	feeCollectorAddr := authtypes.NewModuleAddress(authtypes.FeeCollectorName)
+	balance := k.bankKeeper.GetBalance(sdkCtx, feeCollectorAddr, k.GetParams(sdkCtx).EvmDenom)
+	if balance.IsZero() {
+		return nil
+	}
+
+	chainFeeSplitterAddress := common.HexToAddress(k.GetParams(sdkCtx).ChainFeeSplitterAddress)
+
+	// Check if the chain fee splitter address is the zero address.
+	// In case of zero address, fees are still being collected in the fee collector
+	// module account.
+	if chainFeeSplitterAddress == (common.Address{}) {
+		return nil
+	}
+
+	chainFeeSplitterAddressBytes := chainFeeSplitterAddress.Bytes()
+
+	// Check if the chain fee splitter address is valid format
+	if err := sdk.VerifyAddressFormat(chainFeeSplitterAddressBytes); err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid chain fee splitter address: %s", chainFeeSplitterAddress)
+	}
+
+	// Transfer chain fee to the chain fee splitter contract
+	err := k.bankKeeper.SendCoinsFromModuleToAccount(sdkCtx, authtypes.FeeCollectorName, chainFeeSplitterAddressBytes, sdk.NewCoins(balance))
```

Comment:
Let's add an info log informing about transferring amount X from the fee collector account to the splitter address Y.

Reply:
[99cf088](https://github.com/mezo-org/mezod/pull/417/commits/99cf0889e10a8f96528b92ef8e15f4ff30644a43)